### PR TITLE
fix: health-check should be outside of hub.ts

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

see #9932 

Object Storage code assumed it was safe to health check when creating a hub. It is not, Hubs are created many times. 

## Changes

* does not await storage when creating a hub
* adds object storage to health check

## How did you test this code?

so far... pushing it to CI to see what fails

### CODE REVIEW

**Review by benjackwhite — APPROVED** [review]
lgtm and reflects the discussion outcome well I think

**Review by macobo — APPROVED** [review]
cc @yakkomajuri

File: plugin-server/src/healthcheck.ts [hunk 1]
```diff
@@ -21,6 +22,34 @@ export async function healthcheck(): Promise<boolean> {
...
+            status.error('💔', 'object storage is not healthy')
+            return false
+        }
+    } catch (error) {
+        status.error('💥', 'Object Storage healthcheck: an unexpected error occurred:', error)
+        return false
+    }
+}
+
+export async function healthcheck(): Promise<boolean> {
```
Comment (macobo) [inline plugin-server/src/healthcheck.ts:46]: A mocking unit test would be nice.

Comment (yakkomajuri) [inline plugin-server/src/healthcheck.ts:46]: We're not currently using this even - but happy to update it for when we do use it.

Also note that the idea of this healthcheck is that we might take the task down if it doesn't pass - is that what we wish to do with object storage now?

File: plugin-server/src/main/services/object_storage.ts [hunk 2]
```diff
@@ -9,10 +9,12 @@ export interface ObjectStorage {
     healthCheck: () => Promise<boolean>
 }
 
+// Object Storage added without any uses to flush out deployment concerns.
+// see https://github.com/PostHog/posthog/pull/9901
 export const connectObjectStorage = (serverConfig: Partial<PluginsServerConfig>): ObjectStorage => {
     let storage = {
         healthCheck: async () => {
```
Comment (yakkomajuri) [inline plugin-server/src/main/services/object_storage.ts:16]:
```suggestion
        healthcheck: async () => {
```

File: plugin-server/src/main/services/object_storage.ts [hunk 3]
```diff
@@ -32,27 +34,29 @@ export const connectObjectStorage = (serverConfig: Partial<PluginsServerConfig>)
...
-                    await S3.headBucket({
-                        Bucket: OBJECT_STORAGE_BUCKET,
-                    }).promise()
-                    return true
-                } catch (error) {
-                    return false
-                }
-            },
+            storage = {
+                healthCheck: async () => {
```
Comment (yakkomajuri) [inline plugin-server/src/main/services/object_storage.ts:39]:
```suggestion
                healthcheck: async () => {
```

File: plugin-server/src/main/services/object_storage.ts [hunk 4]
```diff
@@ -32,27 +34,29 @@ export const connectObjectStorage = (serverConfig: Partial<PluginsServerConfig>)
...
+                        return false
+                    }
+
+                    try {
+                        await S3.headBucket({
+                            Bucket: OBJECT_STORAGE_BUCKET,
+                        }).promise()
+                        return true
+                    } catch (error) {
+                        status.error('💣', 'could not access bucket:', error)
```
Comment (yakkomajuri) [inline plugin-server/src/main/services/object_storage.ts:51]:
```suggestion
                        status.error('💣', 'Could not access bucket:', error)
```

**Comment by pauldambra** [comment]
hardly any changes... re-requested review since had to edit a file outside of the change :)

**Comment by pauldambra** [comment]
@yakkomajuri @macobo ping for a final check, very little changed since last review.

Feels like we should get this in ahead of code freeze

**Pull request merged**